pkg/controller/dataimportcron-controller.go

```diff
@@ -271,7 +281,7 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 	if err := r.client.Update(ctx, dataImportCron); err != nil {
 		return res, err
 	}
-	return res, nil
+	return res, r.checkFailingDataImportCrons()
```

Comment:
Isn't it to too much work to scan all the DataImportCrons on each singe DataImportCron update?
Returning checkFailingDataImportCrons() as the error value of the caller function seems a bit strange here.
I would do it right after the conditions update, and consider if we really want to return an error or just warn when it fails.

Reply:
Good point
I have a draft where I tried the incremental way (do it right after the conditions update) but it looks less readable especially in the parts where we have to decrement the metric value

Do you think the list call is that bad, even though it's cached?

Reply:
see my comment on doing it once in the Reconcile